src/lib.rs

```diff
@@ -75,7 +75,30 @@ use tracing;
 /// The client returned by `connect` function
 ///
 /// This is a convenience type which you most likely want to use instead of raw client.
-pub type Client = rpc::lightning_client::LightningClient<tonic::codegen::InterceptedService<tonic::transport::Channel, MacaroonInterceptor>>;
+pub struct Client {
+    inner: rpc::lightning_client::LightningClient<tonic::codegen::InterceptedService<tonic::transport::Channel, MacaroonInterceptor>>,
+
+    wallet: walletrpc::wallet_kit_client::WalletKitClient<tonic::codegen::InterceptedService<tonic::transport::Channel, MacaroonInterceptor>>,
+}
+
+impl Client {
+    pub fn wallet(&mut self) -> &mut walletrpc::wallet_kit_client::WalletKitClient<tonic::codegen::InterceptedService<tonic::transport::Channel, MacaroonInterceptor>> {
+        &mut self.wallet
+    }
+}
+
+impl std::ops::Deref for Client {
+    type Target = rpc::lightning_client::LightningClient<tonic::codegen::InterceptedService<tonic::transport::Channel, MacaroonInterceptor>>;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl std::ops::DerefMut for Client {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.inner
+    }
+}
```

Comment:
This looks like an attempt to do inheritance which is [considered an anti-pattern in Rust](https://rust-unofficial.github.io/patterns/anti_patterns/deref.html) and I agree it's an anti-pattern.

Since there are so many methods, I guess we will have to break the API (note that `Deref` isn't backwards-compatible anyway) and provide lighting via a `lightning` method. But if you feel like manually delegating all methods I'm fine with it.

Reply:
providing lightning via `lightning` method